{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import pandas"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set up server URL\n",
    "Change the URL here to point the client to other implementations of the api.  The examples here are based on the GTEx server running at:\n",
    "http://gtexportal.org/rnaget\n",
    "\n",
    "GTEx has provided documentaation on their implementation at: http://gtexportal.org/rnaget/docs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "server_params = {'gtex': {'rnaget_url': \"http://gtexportal.org/rnaget\",\n",
    "                          'headers': {'User-Agent': 'python-requests/2.21.0',\n",
    "                                      'Accept-Encoding': 'gzip, deflate',\n",
    "                                      'Accept': '*/*',\n",
    "                                      'Connection': 'keep-alive'}\n",
    "                         }\n",
    "                }\n",
    "\n",
    "# use this to select the target server.  This should be the only knob to turn - everything\n",
    "# from here on should use the API.  Your server parameters can be added to the dictionary.\n",
    "rnaget_params = server_params['gtex']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Retrieve implementation details\n",
    "The /service-info endpoint will return useful information about the implementation including the API version implemented and the supported endpoints."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'id': 'org.gtexportal.api.rnaget', 'name': 'GTEx Portal RNAget', 'version': '1.0.0', 'type': {'group': 'org.ga4gh', 'artifact': 'rnaget', 'version': '1.2.0'}, 'organization': {'name': 'GTEx Project', 'url': 'https://gtexportal.org'}, 'description': 'This service provides access to GTEx public RNA-Seq data.', 'contactUrl': 'https://gtexportal.org/home/contact', 'documentationUrl': 'https://gtexportal.org/rnaget/docs', 'createdAt': None, 'updatedAt': None, 'environment': 'prod', 'supported': {'projects': True, 'studies': True, 'expressions': True, 'continuous': False}}\n"
     ]
    }
   ],
   "source": [
    "r = requests.get(f\"{rnaget_params['rnaget_url']}/service-info\", headers=rnaget_params['headers'])\n",
    "print(r.json())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From the service-info we see that v1.2.0 of the API is implemented.  This server also supports the /projects, /studies and /expressions endpoints."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Project"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'id': 'gtex', 'version': '1.0', 'name': 'GTEx', 'description': 'GTEx Public includes all publicly available data generated by the Genotype Tissue Expression (GTEx) Project'}]\n"
     ]
    }
   ],
   "source": [
    "r = requests.get(f\"{rnaget_params['rnaget_url']}/projects\", headers=rnaget_params['headers'])\n",
    "print(r.json())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With the projectID we can get details"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'id': 'gtex', 'version': '1.0', 'name': 'GTEx', 'description': 'GTEx Public includes all publicly available data generated by the Genotype Tissue Expression (GTEx) Project'}\n"
     ]
    }
   ],
   "source": [
    "projectID = 'gtex'\n",
    "r = requests.get(f\"{rnaget_params['rnaget_url']}/projects/{projectID}\", headers=rnaget_params['headers'])\n",
    "print(r.json())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Study\n",
    "The /studies endpoint behaves similarly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'id': 'gtex_v8', 'version': '1.0', 'name': 'GTEx Analysis V8', 'description': 'This is the V8 release of GTEx data. V8 is a full release', 'parentProjectId': 'gtex', 'genome': 'GRCh38/hg38'}]\n"
     ]
    }
   ],
   "source": [
    "r = requests.get(f\"{rnaget_params['rnaget_url']}/studies\", headers=rnaget_params['headers'])\n",
    "print(r.json())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'id': 'gtex_v8', 'version': '1.0', 'name': 'GTEx Analysis V8', 'description': 'This is the V8 release of GTEx data. V8 is a full release', 'parentProjectId': 'gtex', 'genome': 'GRCh38/hg38'}\n"
     ]
    }
   ],
   "source": [
    "studyID = 'gtex_v8'\n",
    "r = requests.get(f\"{rnaget_params['rnaget_url']}/studies/{studyID}\", headers=rnaget_params['headers'])\n",
    "print(r.json())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Expression\n",
    "This is the heart of the RNAget API and allows for several operations.  These examples continue to use the GTEx implementation.  RNAget is a retrieval API so search and discovery are out of scope.  The GTEx documentation does inform us of the available values for the required path parameter expressionId: gene_tpms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'units': 'tsv', 'url': 'https://storage.cloud.google.com/prod_rnaget_working_storage/tmp5nr_zoy8.tsv', 'version': None, 'fileType': None, 'studyID': None, 'headers': None, 'MD5': None}\n"
     ]
    }
   ],
   "source": [
    "# to demonstrate a key API feature and make the example smaller we will request a slice of the matrix\n",
    "featureList = 'EGFR'\n",
    "payload = {\"featureNameList\": featureList, \"tissueSiteDetailIDs\": \"Adipose_Subcutaneous\"}\n",
    "expressionId = 'gene_tpms'\n",
    "r = requests.get(f\"{rnaget_params['rnaget_url']}/expressions/{expressionId}/ticket\", params=payload,\n",
    "                 headers=rnaget_params['headers'])\n",
    "print(r.json())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We have used the /ticket endpoint so the return json includes the URL to download the actual data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Working with the files\n",
    "In addition to getting a download URL, the API has the /bytes endpoint for inline data.  Using this we can stream the expression file to a local file and run whatever downstream analysis we wish.  The following are some simple navigation examples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# to demonstrate a key API feature and make the example smaller we will request a slice of the matrix\n",
    "featureList = ['EGFR','PTEN']\n",
    "payload = {\"featureNameList\": featureList, \"tissueSiteDetailIDs\": \"Adipose_Subcutaneous\"}\n",
    "expressionId = 'gene_tpms'\n",
    "\n",
    "# use the /bytes endpoint to stream the data\n",
    "r = requests.get(f\"{rnaget_params['rnaget_url']}/expressions/{expressionId}/bytes\", params=payload,\n",
    "                 headers=rnaget_params['headers'], stream=True)\n",
    "\n",
    "# save the streamed data locally (to a file in this case)\n",
    "demo_data_file = \"adipose_sub_EGFR_PTEN.tsv\"\n",
    "with open(demo_data_file, 'wb') as fd:\n",
    "    for chunk in r.iter_content(chunk_size=128):\n",
    "        fd.write(chunk)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The tsv file can now be read into whatever dataframe you want using all the usual import mechanisms.  Here pandas is used as an example."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GTEX-1117F-0226-SM-5GZZ7</th>\n",
       "      <th>GTEX-111CU-1826-SM-5GZYN</th>\n",
       "      <th>GTEX-111FC-0226-SM-5N9B8</th>\n",
       "      <th>GTEX-111VG-2326-SM-5N9BK</th>\n",
       "      <th>GTEX-111YS-2426-SM-5GZZQ</th>\n",
       "      <th>GTEX-1122O-2026-SM-9YFMG</th>\n",
       "      <th>GTEX-1128S-2126-SM-5H12U</th>\n",
       "      <th>GTEX-113IC-0226-SM-5HL5C</th>\n",
       "      <th>GTEX-117YX-2226-SM-5EGJJ</th>\n",
       "      <th>GTEX-11DXW-0326-SM-5H11W</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZXES-2026-SM-5NQ6R</th>\n",
       "      <th>GTEX-ZXG5-0226-SM-59HJI</th>\n",
       "      <th>GTEX-ZYFC-0326-SM-5NQ7H</th>\n",
       "      <th>GTEX-ZYFD-0226-SM-5NQ86</th>\n",
       "      <th>GTEX-ZYT6-0326-SM-7LG5R</th>\n",
       "      <th>GTEX-ZYVF-0226-SM-5GIEG</th>\n",
       "      <th>GTEX-ZYW4-0226-SM-5E44M</th>\n",
       "      <th>GTEX-ZYY3-0226-SM-5E45M</th>\n",
       "      <th>GTEX-ZZ64-1626-SM-5E43W</th>\n",
       "      <th>GTEX-ZZPU-2726-SM-5NQ8O</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencode_ids</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000146648.17</th>\n",
       "      <td>82.49</td>\n",
       "      <td>44.13</td>\n",
       "      <td>37.12</td>\n",
       "      <td>16.52</td>\n",
       "      <td>42.25</td>\n",
       "      <td>30.76</td>\n",
       "      <td>41.34</td>\n",
       "      <td>33.35</td>\n",
       "      <td>68.27</td>\n",
       "      <td>33.0</td>\n",
       "      <td>...</td>\n",
       "      <td>46.05</td>\n",
       "      <td>66.89</td>\n",
       "      <td>22.35</td>\n",
       "      <td>28.32</td>\n",
       "      <td>46.09</td>\n",
       "      <td>49.73</td>\n",
       "      <td>59.95</td>\n",
       "      <td>31.19</td>\n",
       "      <td>49.65</td>\n",
       "      <td>56.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000171862.10</th>\n",
       "      <td>42.55</td>\n",
       "      <td>39.73</td>\n",
       "      <td>37.82</td>\n",
       "      <td>19.66</td>\n",
       "      <td>55.36</td>\n",
       "      <td>38.11</td>\n",
       "      <td>33.11</td>\n",
       "      <td>40.25</td>\n",
       "      <td>67.74</td>\n",
       "      <td>25.5</td>\n",
       "      <td>...</td>\n",
       "      <td>37.73</td>\n",
       "      <td>45.91</td>\n",
       "      <td>29.62</td>\n",
       "      <td>48.67</td>\n",
       "      <td>28.54</td>\n",
       "      <td>34.83</td>\n",
       "      <td>53.31</td>\n",
       "      <td>31.31</td>\n",
       "      <td>92.12</td>\n",
       "      <td>46.30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 663 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    GTEX-1117F-0226-SM-5GZZ7  GTEX-111CU-1826-SM-5GZYN  \\\n",
       "gencode_ids                                                              \n",
       "ENSG00000146648.17                     82.49                     44.13   \n",
       "ENSG00000171862.10                     42.55                     39.73   \n",
       "\n",
       "                    GTEX-111FC-0226-SM-5N9B8  GTEX-111VG-2326-SM-5N9BK  \\\n",
       "gencode_ids                                                              \n",
       "ENSG00000146648.17                     37.12                     16.52   \n",
       "ENSG00000171862.10                     37.82                     19.66   \n",
       "\n",
       "                    GTEX-111YS-2426-SM-5GZZQ  GTEX-1122O-2026-SM-9YFMG  \\\n",
       "gencode_ids                                                              \n",
       "ENSG00000146648.17                     42.25                     30.76   \n",
       "ENSG00000171862.10                     55.36                     38.11   \n",
       "\n",
       "                    GTEX-1128S-2126-SM-5H12U  GTEX-113IC-0226-SM-5HL5C  \\\n",
       "gencode_ids                                                              \n",
       "ENSG00000146648.17                     41.34                     33.35   \n",
       "ENSG00000171862.10                     33.11                     40.25   \n",
       "\n",
       "                    GTEX-117YX-2226-SM-5EGJJ  GTEX-11DXW-0326-SM-5H11W  ...  \\\n",
       "gencode_ids                                                             ...   \n",
       "ENSG00000146648.17                     68.27                      33.0  ...   \n",
       "ENSG00000171862.10                     67.74                      25.5  ...   \n",
       "\n",
       "                    GTEX-ZXES-2026-SM-5NQ6R  GTEX-ZXG5-0226-SM-59HJI  \\\n",
       "gencode_ids                                                            \n",
       "ENSG00000146648.17                    46.05                    66.89   \n",
       "ENSG00000171862.10                    37.73                    45.91   \n",
       "\n",
       "                    GTEX-ZYFC-0326-SM-5NQ7H  GTEX-ZYFD-0226-SM-5NQ86  \\\n",
       "gencode_ids                                                            \n",
       "ENSG00000146648.17                    22.35                    28.32   \n",
       "ENSG00000171862.10                    29.62                    48.67   \n",
       "\n",
       "                    GTEX-ZYT6-0326-SM-7LG5R  GTEX-ZYVF-0226-SM-5GIEG  \\\n",
       "gencode_ids                                                            \n",
       "ENSG00000146648.17                    46.09                    49.73   \n",
       "ENSG00000171862.10                    28.54                    34.83   \n",
       "\n",
       "                    GTEX-ZYW4-0226-SM-5E44M  GTEX-ZYY3-0226-SM-5E45M  \\\n",
       "gencode_ids                                                            \n",
       "ENSG00000146648.17                    59.95                    31.19   \n",
       "ENSG00000171862.10                    53.31                    31.31   \n",
       "\n",
       "                    GTEX-ZZ64-1626-SM-5E43W  GTEX-ZZPU-2726-SM-5NQ8O  \n",
       "gencode_ids                                                           \n",
       "ENSG00000146648.17                    49.65                    56.88  \n",
       "ENSG00000171862.10                    92.12                    46.30  \n",
       "\n",
       "[2 rows x 663 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pandas.read_csv(demo_data_file, sep='\\t', header=0, index_col=0)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ENSG00000146648.17', 'ENSG00000171862.10'], dtype='object', name='gencode_ids')"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All the usual dataframe actions can be done.  Like selecting a subset:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GTEX-1117F-0226-SM-5GZZ7</th>\n",
       "      <th>GTEX-111CU-1826-SM-5GZYN</th>\n",
       "      <th>GTEX-111FC-0226-SM-5N9B8</th>\n",
       "      <th>GTEX-111VG-2326-SM-5N9BK</th>\n",
       "      <th>GTEX-111YS-2426-SM-5GZZQ</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gencode_ids</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000146648.17</th>\n",
       "      <td>82.49</td>\n",
       "      <td>44.13</td>\n",
       "      <td>37.12</td>\n",
       "      <td>16.52</td>\n",
       "      <td>42.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000171862.10</th>\n",
       "      <td>42.55</td>\n",
       "      <td>39.73</td>\n",
       "      <td>37.82</td>\n",
       "      <td>19.66</td>\n",
       "      <td>55.36</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    GTEX-1117F-0226-SM-5GZZ7  GTEX-111CU-1826-SM-5GZYN  \\\n",
       "gencode_ids                                                              \n",
       "ENSG00000146648.17                     82.49                     44.13   \n",
       "ENSG00000171862.10                     42.55                     39.73   \n",
       "\n",
       "                    GTEX-111FC-0226-SM-5N9B8  GTEX-111VG-2326-SM-5N9BK  \\\n",
       "gencode_ids                                                              \n",
       "ENSG00000146648.17                     37.12                     16.52   \n",
       "ENSG00000171862.10                     37.82                     19.66   \n",
       "\n",
       "                    GTEX-111YS-2426-SM-5GZZQ  \n",
       "gencode_ids                                   \n",
       "ENSG00000146648.17                     42.25  \n",
       "ENSG00000171862.10                     55.36  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.iloc[:,:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "and all of the usual statistics that Pandas defines."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000146648.17\n",
      "Mean: 41.620633484162894\n",
      "Std Dev: 16.43412996464274\n",
      "\n",
      "ENSG00000171862.10\n",
      "Mean: 47.18760180995475\n",
      "Std Dev: 17.90503536354805\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for i in range(df.index.size):\n",
    "    print(f'{df.index[i]}')\n",
    "    print(f'Mean: {df.iloc[i, :].mean()}\\nStd Dev: {df.iloc[i, :].std()}\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Wrap it up\n",
    "We can tie it all together in a nice little function that uses RNAget to query the GTEx server for the data to calculate summary statistics on a list of genes for a provided tissue."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def summarize_gtex_tissue_expression(tissue, expression_id, studyID, genelist=None, localfile=None):\n",
    "    payload = {\"tissueSiteDetailIDs\": tissue}\n",
    "    if genelist is not None:\n",
    "        payload['featureNameList'] = genelist\n",
    "\n",
    "    # use the /bytes endpoint to stream the data\n",
    "    r = requests.get(f\"{rnaget_params['rnaget_url']}/expressions/{expressionId}/bytes\", params=payload,\n",
    "                     headers=rnaget_params['headers'], stream=True)\n",
    "\n",
    "    if localfile is not None:\n",
    "        # save the streamed data locally (to a file in this case)\n",
    "        with open(localfile, 'wb') as fd:\n",
    "            for chunk in r.iter_content(chunk_size=128):\n",
    "                fd.write(chunk)\n",
    "\n",
    "        df = pandas.read_csv(localfile, sep='\\t', header=0, index_col=0)\n",
    "    # the else, just stream it into the df is left as an exercise to implement\n",
    "\n",
    "    s = requests.get(f\"{rnaget_params['rnaget_url']}/studies/{studyID}\", headers=rnaget_params['headers'])\n",
    "    gtex_name = s.json()['name']\n",
    "\n",
    "    print(f'Gene Summary Statistics for {tissue}\\nStudy: {gtex_name}')\n",
    "    for i in range(df.index.size):\n",
    "        print(f'{df.index[i]}')\n",
    "        print(f'Mean: {df.iloc[i, :].mean()}\\nStd Dev: {df.iloc[i, :].std()}\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene Summary Statistics for Adipose_Subcutaneous\n",
      "Study: GTEx Analysis V8\n",
      "ENSG00000146648.17\n",
      "Mean: 41.620633484162894\n",
      "Std Dev: 16.43412996464274\n",
      "\n",
      "ENSG00000171862.10\n",
      "Mean: 47.18760180995475\n",
      "Std Dev: 17.90503536354805\n",
      "\n"
     ]
    }
   ],
   "source": [
    "featureList = ['EGFR','PTEN']\n",
    "demo_data_file = \"adipose_sub_EGFR_PTEN.tsv\"\n",
    "tissue = \"Adipose_Subcutaneous\"\n",
    "expression_id = 'gene_tpms'\n",
    "studyID = 'gtex_v8'\n",
    "summarize_gtex_tissue_expression(tissue, expression_id, studyID, genelist=featureList, localfile=demo_data_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
